reversion/backends/dynamodb/queryset.py

```diff
@@ -63,7 +63,7 @@ def _process_execution_with_prefetch_prev_version(self):
                     version.prev_version = prev_version
                 if len(results) > self._limit:
                     self._next_key = {
-                        key: self._execution._items[len(self._results) - 1][key]
+                        key: self._results[-1].serialize()[key]
```

Comment:
pyston result iterator uses inner pagination therefore if self._results has 20 items self._execution._items can contain only 16 items (because first 4 items was loaded in the first page). Therefore I must use _results and serialize its items to the dict form to get last _next_key